Class loading issues with JSR-223 Groovy ScriptEngine

It looks like the classes from `org.eclipse.smarthome.automation.module.script.rulesupport.shared.simple` are not on the classpath of the GroovyScriptEngine because importing or extending `SimpleRule` results in class loading issues.


To reproduce:
1. Download https://dl.bintray.com/groovy/maven/apache-groovy-binary-2.4.12.zip
2. Extract  `/groovy-2.4.12/lib/groovy*.jar` to `/runtime/lib/ext`
3. Start runtime and install Rule Engine (Experimental) (Paper UI -> Add-ons -> Misc)
4. Add test.groovy file to `/conf/automation/jsr223` with the following contents:

```
import org.eclipse.smarthome.automation.*
import org.eclipse.smarthome.automation.module.script.rulesupport.shared.simple.*

scriptExtension.importPreset("RuleSupport")
scriptExtension.importPreset("RuleSimple")


class TestRule extends SimpleRule {
    Object execute(Action module, Map<String, ?> inputs) {
        println("testRule.execute()")
    }
}

def rule = new TestRule()

automationManager.addRule(rule)
```

The engine loads the script but the logging now shows:

```
10:23:41.258 [INFO ] [rt.internal.loader.ScriptFileWatcher] - Loading script 'test.groovy'
10:23:41.267 [ERROR] [ipt.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/etc/openhab2/automation/jsr223/test.groovy': org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
Script21.groovy: 8: unable to resolve class SimpleRule 
 @ line 8, column 1.
   class TestRule extends SimpleRule {
   ^

1 error
```

----

It looks like other packages are on the Groovy classpath because it is possible to add a rule via Groovy without using the `SimpleRule` e.g.:

```
import org.eclipse.smarthome.automation.*

scriptExtension.importPreset("RuleSupport")
scriptExtension.importPreset("RuleSimple")


class TestRule extends Rule {
}

def rule = new TestRule()

automationManager.addRule(rule)
```

--- a/bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/META-INF/MANIFEST.MF
+++ b/bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/META-INF/MANIFEST.MF
@@ -8,6 +8,9 @@ Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Import-Package: org.eclipse.smarthome.automation,
  org.eclipse.smarthome.automation.handler,
  org.eclipse.smarthome.automation.module.script,
+ org.eclipse.smarthome.automation.module.script.rulesupport.shared,
+ org.eclipse.smarthome.automation.module.script.rulesupport.shared.factories,
+ org.eclipse.smarthome.automation.module.script.rulesupport.shared.simple,
  org.eclipse.smarthome.automation.parser,
  org.eclipse.smarthome.automation.template,
  org.eclipse.smarthome.automation.type,
@@ -27,6 +30,9 @@ Import-Package: org.eclipse.smarthome.automation,
  org.osgi.service.component,
  org.osgi.util.tracker,
  org.slf4j
+Export-Package: org.eclipse.smarthome.automation.module.script.rulesupport.shared,
+ org.eclipse.smarthome.automation.module.script.rulesupport.shared.factories,
+ org.eclipse.smarthome.automation.module.script.rulesupport.shared.simple
 Automation-ResourceType: json
 Service-Component: OSGI-INF/*.xml
 Bundle-ActivationPolicy: lazy

